components/button/floating-buttons.js

```diff
@@ -270,6 +276,27 @@ class FloatingButtons extends RtlMixin(LitElement) {
 		});
 	}
 
+	_scrollIfFloatObsuringFocus() {
+
+		if (!this._floating) return;
+
+		const currentFocusedItem = getComposedActiveElement();
```

Comment:
@awikkerink I found that `document.activeElement` can return parent custom elements in certain scenarios and that was messing with the height calculations. This ensures that it's always the actual element with focus.